andela/travel_tool_back
# #161374998 View travel calendar showing itinerary of travellers

#### What does this PR do?
This PR contains the implementation of allowing a travel admin to view travel calendar showing itinerary of travellers to and from his/her centre

#### Description of Task to be completed?
- Get flight/itinerary details of travellers.
- Implement pagination for the list of details gotten from the db.

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_back.git`
- Enter into the project directory - `cd travel_tool_back`
- checkout the branch `ft-see-travel-calender-161374998`
- Install dependencies - `yarn install`
- Rollback and run database migrations - `yarn db:rollmigrate`
- Start the server - `yar start`
- Navigate to [travela](http://travela-local.andela.com:3000/#) in your browser and login
- Login on your browser and grab your token from the application tab in Developers tab
- Open Postman
- Add your token in headers
- Nagivate to your database GUI (pgAdmin or postico)
- Update your user role in UserRoles table to 'Travel Administrator' `29187`
- In postman make a get Request to the endpoint: ```http://localhost:5000/api/v1/analytics/calendar?dateFrom=2018-10-02&dateTo=2018-12-30&type=json&limit=5&page=1&location=Nairobi, Kenya```
- You should get an error message 'Flight Details not Found'

##### Create Requests
- Update your user role in UserRoles table to 'Manage' `53019`
- Create a new one-way trip request
- Approve the request
- Add Ticket details in the Travel checklist for your request
- Update your user role in UserRoles table to 'Travel Administrator' `29187`
- In postman make a get Request to the endpoint: ```http://localhost:5000/api/v1/analytics/calendar```
and provide the parameters
type, location, dateFrom, dateTo, limit and page as shown below
``` http://localhost:5000/api/v1/analytics/calendar?dateFrom=2018-10-02&dateTo=2018-12-30&type=json&limit=5&page=1&location=Nairobi, Kenya```
- Repeat the process to test for other trip types

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[161374998](https://www.pivotaltracker.com/story/show/161374998)

#### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2018-11-21 at 19 10 15" src="https://user-images.githubusercontent.com/13402544/48853721-2f219080-edc1-11e8-914b-a8db8f432f1f.png">

#### Questions:
